### PR TITLE
Feature/init db

### PR DESCRIPTION
**Pull request by JorisPineau**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [ ] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I ran `bun run format`/`bun run lint` before commit
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation where needed
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings
- [ ] I have updated `develop` and merged to my branch before submitting pull request
- [ ] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by Abecker37 — DISMISSED** [review]

**Review by herve-perel — APPROVED** [review]

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  // Migration code that reverts the database to the previous state.
+  await db.schema.dropTable('user').ifExists().execute();
+  await db.schema.dropTable('trip').ifExists().execute();
+  await db.schema.dropTable('car').ifExists().execute();
+  await db.schema.dropTable('car_type').ifExists().execute();
+  await db.schema.dropTable('reservation').ifExists().execute();
+  await db.schema.dropTable('reservation_seat').ifExists().execute();
+  await db.schema.dropTable('checkpoint_trip').ifExists().execute();
+  await db.schema.dropTable('notice').ifExists().execute();
+  await db.schema.dropTable('messaging').ifExists().execute();
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:17]: Il faut inverser
Les tables créées en dernier doivent être supprimées en premier

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('driver_id', 'int8', (col) =>
+      col.references('driver.id').notNull().unsigned(),
+    )
+    .addColumn('created_at', 'datetime', (col) => col.notNull())
+    .addColumn('date', 'datetime', (col) => col.notNull())
+    .addColumn('kilometer', 'decimal(6, 1)', (col) => col.notNull().unsigned())
+    .addColumn('travel_time', 'datetime', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:55]: La durée du trajet ne peut pas être une date

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('seat_available', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('price', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('is_auto_validation', 'boolean', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:58]:
```suggestion
    .addColumn('should_auto_validate', 'boolean', (col) => col.notNull())
```
"est auto validation" -> "devrait auto valider"

File: packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    .addColumn('comment', 'text')
+    .addColumn('is_baby_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_non_vaccinated_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_animal_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_smoker_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_with_toll', 'boolean', (col) => col.notNull())
```
Comment (YDrogen) [inline packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts:64]:
```suggestion
    .addColumn('has_tolls', 'boolean', (col) => col.notNull())
```
"est avec péage" -> "a (des) péages"

File: packages/types/src/database.ts [hunk 5]
```diff
@@ -1,14 +1,137 @@
...
+  travel_time: Date;
+  seat_available: number;
+  price: number;
+  is_auto_validation: boolean;
+  comment?: string;
+  is_baby_allowed: boolean;
+  is_non_vaccinated_allowed: boolean;
+  is_animal_allowed: boolean;
+  is_smoker_allowed: boolean;
+  is_with_toll: boolean;
```
Comment (YDrogen) [inline packages/types/src/database.ts:36]: bien penser à mettre à jour ça

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by herve-perel — DISMISSED** [review]

**Review by herve-perel — DISMISSED** [review]

**Review by YDrogen — CHANGES_REQUESTED** [review]
Les migrations ne fonctionnent pas

**Review by herve-perel — APPROVED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by herve-perel — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Review by Abecker37 — APPROVED** [review]

**Pull request merged**